eolearn/mask/cloud_mask.py

```diff
@@ -41,6 +41,159 @@ def predict_proba(self, X: np.ndarray) -> np.ndarray:  # noqa: N803
 
 
 class CloudMaskTask(EOTask):
+    """Cloud masking with the s2cloudless model. Outputs a cloud mask, while the output of cloud probabilities
+    is optional.
+
+    Example usage:
+    .. code-block:: python
+        task = CloudMaskTask(
+            data_feature = (FeatureType.DATA, 'BANDS'),
+            valid_data_feature = (FeatureType.MASK, 'IS_DATA'),
+            output_mask_feature=(FeatureType.MASK, 'CLM_S2C'),
+            output_proba_feature=(FeatureType.DATA, 'CLP_S2C')
+        )
+    """
+
+    MODELS_FOLDER = os.path.join(os.path.dirname(__file__), "models")
+    CLASSIFIER_NAME = "pixel_s2_cloud_detector_lightGBM_v0.2.txt"
+
+    def __init__(
+        self,
+        data_feature: tuple[FeatureType, str],
+        valid_data_feature: tuple[FeatureType, str],
+        output_mask_feature: tuple[FeatureType, str],
+        output_proba_feature: tuple[FeatureType, str] | None = None,
+        data_indices: list[int] | None = None,
+        threshold: float = 0.4,
+        average_over: int | None = 4,
+        dilation_size: int | None = 2,
+    ):
+        """
+        :param data_feature: A data feature which stores raw Sentinel-2 reflectance bands.
+        :param valid_data_feature: A mask feature which indicates whether data is valid.
+        :param output_mask_feature: The output feature containing cloud masks.
+        :param output_proba_feature: The output feature containing cloud probabilities. By default this is not saved.
+        :param data_indices: List of indices to use in case of custom input data. Defaults to 10 band indices used in
+            s2cloudless '("B01", "B02", "B04", "B05", "B08", "B8A", "B09", "B10", "B11", "B12")'.
+        :param threshold: Cloud probability threshold for the classifier. Defaults to `0.4`.
+        :param average_over: Size of the pixel neighbourhood used in the averaging post-processing step.
+            A value of `0` skips this post-processing step. Default value mimics the default for
+            s2cloudless: `4`.
+        :param dilation_size: Size of the dilation post-processing step. A value of `0` or `None` skips this
+            post-processing step. Default value mimics the default for s2cloudless: `2`.
```

Comment:
I would argue that `None` should be the only 'skip' value

Reply:
I think this is meant as "practically skips", because dilation of 0 does basically nothing. But I guess the process is still executed. Will just update the description that `None` should be provided to skip.

Reply:
I checked, the `> 0` condition is also in the code, but I think it's fine if we omit this from the description.